plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -370,6 +383,14 @@ newtype CekCarryingM term uni fun s a = CekCarryingM
     { unCekCarryingM :: ST s a
     } deriving newtype (Functor, Applicative, Monad)
 
+instance Show (CekCarryingM term uni fun s UTCTime) where
+    show (CekCarryingM x) = show x
+
+-- We need this show instance to show the UTCTime properly.
+-- The default Show (ST s a) instance is the string "ST actions".
+instance {-# OVERLAPS #-} Show (ST s UTCTime) where
+    showsPrec _ x  = showString $ show x
```

Comment:
This seems very strange. We should never need to show a value of type `ST s UTCTime`, but rather a value of type `UTCTime`.

Reply:
As the code was, without the instance the log showed "ST actions" instead of the time stamp. With the instance, the log showed the UTC time. It may be fine with the updated code. I'll know when I test it more.

Reply:
I mean, in `showsPrec _ x  = showString $ show x` you're calling `show` over the same `x` that `showsPrec` receives. `show x` literally unfolds into `showsPrec 0 x ""`, so does that even terminate?

Or if you don't trust this reasoning, the following also type checks:

```haskell
data NoShow

instance {-# OVERLAPS #-} Show (ST s NoShow) where
    showsPrec _ x  = showString $ show x
``` 

`NoShow` has no `Show` instance and yet the `Show (ST s NoShow)` instance type checks hence it can't defer to the non-existent `Show` instance of `NoShow` and hence the same does not happen for `UTCTime`.

Reply:
It did terminate and showed the time. But Michael is right I should have taken it out of the `ST` and not need this `Show (ST s UTCTime)` instance.

Reply:
It did get stuck in an earlier version but then it did terminate and showed the time. But Michael is right I should have taken it out of the ST and not need this Show (ST s UTCTime) instance.

Reply:
> It did get stuck in an earlier version but then it did terminate and showed the time.

My comment was about the version that is shown above. It does not terminate. Here I went to the trouble of checking that.

![Screenshot from 2021-08-04 00-16-21](https://user-images.githubusercontent.com/10480926/128087427-df083c52-8fd3-4a5a-a40d-e7c85997b89f.png)

Reply:
Thanks for letting me know. I ran `cabal test plutus-tx-tests` and looked at the errors and the output log instead. I'm not sure why the time showed up there. Maybe I got confused somehow. Anyway I'll check with what you did here next time too. Also I realized I should avoid writing overlapping custom instance in general! 🤓